pkg/network/read_buffer_amd64.go

```diff
@@ -0,0 +1,23 @@
+// +build windows
+
+package network
+
+// This file contains AMD64 specific implementations for allocating and freeing readbuffers. AMD64
+// is the only supported architecture, but the AMD64 implementation will not
+// compiled on i386 because the signature for malloc changes based on bus size.
+
+// #include <stdlib.h>
+import "C"
+
+import (
+	"unsafe"
+)
+
+func allocateReadBuffer() (*_readbuffer, error) {
+	sizeOfReadBuffer := unsafe.Sizeof(_readbuffer{})
+	return (*_readbuffer)(C.malloc(C.ulonglong(sizeOfReadBuffer))), nil
```

Comment:
Did you try `C.malloc(C.size_t(unsafe.Sizeof(_readbuffer{})))` to avoid the i386 problem?

Reply:
No, I'll give it a shot